src/query_parser/v2/parser.y

```diff
@@ -733,67 +684,91 @@ text_expr(A) ::= PERCENT PERCENT PERCENT param_term(B) PERCENT PERCENT PERCENT.
 
 modifier(A) ::= MODIFIER(B) . {
   B.len = unescapen((char*)B.s, B.len);
-  A = B;
+  A.tok = B;
+  if (ctx->sctx->spec) {
+    A.fs = IndexSpec_GetFieldWithLength(ctx->sctx->spec, B.s, B.len);
+    if (!A.fs) {
+      reportSyntaxError(ctx->status, &A.tok, "Unknown field");
+    }
+  }
 }
 
 modifierlist(A) ::= modifier(B) OR term(C). {
-  if (__builtin_expect(C.len > 0, 1)) {
-    A = NewVector(char *, 2);
-    char *s = rm_strndup(B.s, B.len);
-    Vector_Push(A, s);
-    s = rm_strndup(C.s, C.len);
-    Vector_Push(A, s);
+  if (ctx->sctx->spec) {
+    if (!FIELD_IS(B.fs, INDEXFLD_T_FULLTEXT)) {
+      REPORT_WRONG_FIELD_TYPE(B, SPEC_TEXT_STR);
+      A = NULL;
+    } else {
+      FieldName second = { .tok = C, .fs = IndexSpec_GetFieldWithLength(ctx->sctx->spec, C.s, C.len) };
+      if (!second.fs) {
+        reportSyntaxError(ctx->status, &second.tok, "Unknown field");
+        A = NULL;
+      } else if (!FIELD_IS(second.fs, INDEXFLD_T_FULLTEXT)) {
+        REPORT_WRONG_FIELD_TYPE(second, SPEC_TEXT_STR);
+        A = NULL;
+      } else {
+        A = array_new(FieldName, 2);
+        array_append(A, B);
+        array_append(A, second);
+      }
+    }
   } else {
-    reportSyntaxError(ctx->status, &C, "Syntax error");
-    A = NULL;
+    A = array_new(FieldName, 2);
+    array_append(A, B);
+    FieldName second = { .tok = C };
+    array_append(A, second);
   }
 }
 
 
 modifierlist(A) ::= modifierlist(B) OR term(C). {
-  if (__builtin_expect(C.len > 0, 1)) {
-    char *s = rm_strndup(C.s, C.len);
-    Vector_Push(B, s);
-    A = B;
-  } else {
-    reportSyntaxError(ctx->status, &C, "Syntax error");
-    if (B) {
-      for (size_t i = 0; i < Vector_Size(B); i++) {
-        char *s;
-        Vector_Get(B, i, &s);
-        rm_free(s);
-      }
-      Vector_Free(B);
+  if (ctx->sctx->spec) {
+    FieldName second = { .tok = C, .fs = IndexSpec_GetFieldWithLength(ctx->sctx->spec, C.s, C.len) };
+    if (!second.fs) {
+      reportSyntaxError(ctx->status, &second.tok, "Unknown field");
+      array_free(B);
+      A = NULL;
+    } else if (!FIELD_IS(second.fs, INDEXFLD_T_FULLTEXT)) {
+      REPORT_WRONG_FIELD_TYPE(second, SPEC_TEXT_STR);
+      array_free(B);
+      A = NULL;
+    } else {
+      A = B;
+      array_append(A, second);
     }
-    A = NULL;
+  } else {
+    A = B;
+    FieldName second = { .tok = C };
+    array_append(A, second);
```

Comment:
When do we get here?

Reply:
Only possible when we don't have a spec context, which should only happen if we parse the query from the coordinator (not for the iterator's purpose)